shared/src/main/scala/mlscript/codegen/Scope.scala

```diff
@@ -301,6 +302,15 @@ class Scope(name: Str, enclosing: Opt[Scope]) {
     * Shorthands for deriving function scopes.
     */
   def derive(name: Str, params: Ls[Str]): Scope = Scope(name, params, this)
+  
+  def refreshRes(): Unit = {
```

Comment:
Why do we need this?

Reply:
I use this function to remove all self-defined `res` variables when they should be overwritten by the auto-`res`.

Reply:
Why not just remove `res`? Is it needed to remove all `res$i` variables?

Reply:
I think you are right. Just resetting the `res` is enough.